hnn_core/dipole.py

```diff
@@ -145,6 +145,101 @@ def simulate_dipole(net, n_trials=1, n_jobs=1):
     return dpl
 
 
+def average_dipoles(dpls):
+    """Compute average over a list of Dipole objects.
+    Parameters
+    ----------
+    dpls: list of Dipole objects
+        Contains list of dipole results to be averaged
+
+    Returns
+    -------
+    dpl: instance of Dipole
+        A dipole object with averages of the dipole data
+    """
+    # need at least on Dipole to get times
+    assert (len(dpls) > 0)
+
+    agg_avg = np.mean(np.array([dpl.dpl['agg'] for dpl in dpls]), axis=0)
+    L5_avg = np.mean(np.array([dpl.dpl['L5'] for dpl in dpls]), axis=0)
+    L2_avg = np.mean(np.array([dpl.dpl['L2'] for dpl in dpls]), axis=0)
+
+    avg_dpl_data = np.c_[agg_avg,
+                         L2_avg,
+                         L5_avg]
+
+    avg_dpl = Dipole(dpls[0].t, avg_dpl_data)
+
+    return avg_dpl
+
+
+def rmse(dpl, exp_dpl, tstart=0.0, tstop=0.0, weights=None):
+    """ Calculates RMSE between data in dpl and exp_dpl
+    Parameters
+    ----------
+    dpl: instance of Dipole
+        A dipole object with simulated data
+    exp_dpl: instance of Dipole
+        A dipole object with experimental data
+    tstart | None: float
+        Time at beginning of range over which to calculate RMSE
+    tstop | None: float
+        Time at end of range over which to calculate RMSE
+    weights | None: array
+        An array of weights to be applied to each point in
+        simulated dpl. Must have length >= dpl.dpl
+        If None, weights will be replaced with 1's for typical RMSE
+        calculation.
+
+    Returns
+    -------
+    err: float
+        Weighted RMSE between data in dpl and exp_dpl
+    """
+    from scipy import signal
+
+    exp_times = exp_dpl.t
+    sim_times = dpl.t
+
+    # do tstart and tstop fall within both datasets?
+    # if not, use the closest data point as the new tstop/tstart
+    for tseries in [exp_times, sim_times]:
+        if tstart < tseries[0]:
+            tstart = tseries[0]
+        if tstop > tseries[-1]:
+            tstop = tseries[-1]
```

Comment:
This is dangerous, you should raise an error here

Reply:
Yeah, it could be surprising behavior, but it's not an error to HNN. If a user adds an arbitrary experimental dipole file, that's fine, but RMSE should only be calculated over the common region. I think warnings.warn might be a better way to communicate to the user.

Reply:
can you describe a use case where it's actually helpful?

Reply:
I had a different example in mind above. Let me start over. Say the user loads in a short experimental data set. Simulation tstop is way beyond the end of the exp data. So this updates tstop be the last data point. The closest data point is still identified below. Once the region of interest has been changed to the smallest common region, then downsampling is performed.

Reply:
Credit to @rythorpe for identifying this case

Reply:
Then you should crop to tstop and then compute rmse.

```py
exp_dipole.crop(tmax=5.0)
rmse(exp_dipole, dipole)
```